app/uk/gov/hmrc/cdsreimbursementclaimfrontend/models/upscan/ScheduledDocument.scala

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cdsreimbursementclaimfrontend.models.upscan
+
+import play.api.libs.json.{Json, OFormat}
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.models.upscan.UpscanCallBack.UpscanSuccess
+
+import java.time.LocalDateTime
+
+final case class ScheduledDocument(
+  fileName: String,
+  uploadReference: UploadReference,
+  uploadedOn: LocalDateTime,
+  upscanUploadMeta: UpscanUploadMeta,
+  upscanSuccess: UpscanSuccess
```

Comment:
is there a reason you didn't include the document type here?

Reply:
We also discussed yesterday to generalise this so that there is only one case class definition to handle the upload. It will always be the same for every type of file upload we need to do for the service so makes sense to generalise it. So something like:

```scala
final case class FileUpload(
  uploadReference: UploadReference,
  upscanUploadMeta: UpscanUploadMeta,
  uploadedOn: LocalDateTime,
  upscanSuccess: UpscanSuccess,
  fileName: String,
  documentType: Option[SupportingEvidenceDocumentType]
)
```

Reply:
The `documentType: Option[SupportingEvidenceDocumentType]` is not part of the general FileUpload

Reply:
Initially - https://github.com/hmrc/cds-reimbursement-claim-frontend/pull/420/commits/95caec685931e4f264d4aa99f116118a8b01ef10
I created a common trait 
```
trait UploadDocument {
  val fileName: String
  val uploadReference: UploadReference
  val uploadedOn: LocalDateTime
  val upscanUploadMeta: UpscanUploadMeta
  val upscanSuccess: UpscanSuccess
}
```
for the Supporting Evidence and Scheduled Document and then removed thinking on revisiting this at the point then I am actually going to be submitting documents in the backend service. 
PR is too huge already even for review

Reply:
> The `documentType: Option[SupportingEvidenceDocumentType]` is not part of the general FileUpload

It actually is but the value should be hardcoded to ScheduleOfMrn (will get the correct name later).

This type here:

```scala
object SupportingEvidenceDocumentType {

  case object C88E2 extends SupportingEvidenceDocumentType
  case object CommercialInvoice extends SupportingEvidenceDocumentType
  case object PackingList extends SupportingEvidenceDocumentType
  case object AirWayBill extends SupportingEvidenceDocumentType
  case object BillOfLading extends SupportingEvidenceDocumentType
  case object SubstituteEntry extends SupportingEvidenceDocumentType
  case object ProofOfAuthority extends SupportingEvidenceDocumentType
  case object CorrespondenceTrader extends SupportingEvidenceDocumentType
  case object Other extends SupportingEvidenceDocumentType
  case object ImportAndExportDeclaration extends SupportingEvidenceDocumentType
```

needs to be renamed to something more general like `UploadDocumentType`

Reply:
> Initially - [95caec6](https://github.com/hmrc/cds-reimbursement-claim-frontend/commit/95caec685931e4f264d4aa99f116118a8b01ef10)
> I created a common trait
> 
> ```
> trait UploadDocument {
>   val fileName: String
>   val uploadReference: UploadReference
>   val uploadedOn: LocalDateTime
>   val upscanUploadMeta: UpscanUploadMeta
>   val upscanSuccess: UpscanSuccess
> }
> ```
> 
> for the Supporting Evidence and Scheduled Document and then removed thinking on revisiting this at the point then I am actually going to be submitting documents in the backend service.
> PR is too huge already even for review

I don't understand - why would you make it a trait from a case class?

Reply:
> The `documentType: Option[SupportingEvidenceDocumentType]` is not part of the general FileUpload

I'm not sure how this has happened but in the BE we have this model:

```scala
sealed trait SupportingEvidenceDocumentType extends Product with Serializable

object SupportingEvidenceDocumentType {
  case object C88E2 extends SupportingEvidenceDocumentType {
    override def toString: String = "C88E2"
  }
  case object CommercialInvoice extends SupportingEvidenceDocumentType {
    override def toString: String = "Commercial Invoice"
  }
  case object PackingList extends SupportingEvidenceDocumentType {
    override def toString: String = "Packing List"
  }
  case object AirWayBill extends SupportingEvidenceDocumentType {
    override def toString: String = "AirWay Bill"
  }
  case object BillOfLading extends SupportingEvidenceDocumentType {
    override def toString: String = "Bill Of Lading"
  }
  case object SubstituteEntry extends SupportingEvidenceDocumentType {
    override def toString: String = "Substitute Entry"
  }
  case object ScheduleOfMRNs extends SupportingEvidenceDocumentType {
    override def toString: String = "Schedule Of MRNs"
  }
  case object ProofOfAuthority extends SupportingEvidenceDocumentType {
    override def toString: String = "Proof Of Authority"
  }
  case object CorrespondenceTrader extends SupportingEvidenceDocumentType {
    override def toString: String = "Correspondence Trader"
  }
  case object AdditionalSupportingDocuments extends SupportingEvidenceDocumentType {
    override def toString: String = "Additional Supporting Documents"
  }
  case object ImportAndExportDeclaration extends SupportingEvidenceDocumentType {
    override def toString: String = "Import and Export Declaration"
  }
  case object Other extends SupportingEvidenceDocumentType {
    override def toString: String = "Other"
  }

  implicit val format: OFormat[SupportingEvidenceDocumentType] = derived.oformat[SupportingEvidenceDocumentType]()
}
```

and in the FE we have this model:

```scala
sealed trait SupportingEvidenceDocumentType extends Product with Serializable

object SupportingEvidenceDocumentType {

  case object C88E2 extends SupportingEvidenceDocumentType
  case object CommercialInvoice extends SupportingEvidenceDocumentType
  case object PackingList extends SupportingEvidenceDocumentType
  case object AirWayBill extends SupportingEvidenceDocumentType
  case object BillOfLading extends SupportingEvidenceDocumentType
  case object SubstituteEntry extends SupportingEvidenceDocumentType
  case object ProofOfAuthority extends SupportingEvidenceDocumentType
  case object CorrespondenceTrader extends SupportingEvidenceDocumentType
  case object Other extends SupportingEvidenceDocumentType
  case object ImportAndExportDeclaration extends SupportingEvidenceDocumentType

  implicit val format: OFormat[SupportingEvidenceDocumentType] = derived.oformat[SupportingEvidenceDocumentType]()

  lazy val SupportingEvidenceDocumentTypes = IndexedSeq(
    C88E2,
    CommercialInvoice,
    PackingList,
    AirWayBill,
    BillOfLading,
    SubstituteEntry,
    ProofOfAuthority,
    CorrespondenceTrader,
    ImportAndExportDeclaration,
    Other
  )
}

```

You can see in the backend we have this `ScheduleOfMRNs` - this is what the supportingevidencedocumentype value should be for schedule of mrn upload document type.